### PR TITLE
Add Japanese version

### LINKED ISSUE #586: Add Japanese version

### PR DESCRIPTION
**Pull request by 38elements**
This resolves #586.
Document for version 8 is not provided in this Japanese version.
`content/ja/about/libraries-addons.md` and `content/ja/about/demos-examples.md` is not included since they are high cost and low effect.
This is a improved and added version of English version.
I asked @eiiches for a review.

### CODE REVIEW

**Comment by developit** [comment]
This is amazing work @38elements!

**Comment by 38elements** [comment]
Review for `cli/` is completed.

**Comment by 38elements** [comment]
Review for following pages is completed.

content/ja/guide/v10/api-reference.md
content/ja/guide/v10/components.md
content/ja/guide/v10/context.md
content/ja/guide/v10/debugging.md

**Comment by 38elements** [comment]
Review for following pages were completed.

content/ja/guide/v10/differences-to-react.md
content/ja/guide/v10/external-dom-mutations.md

Review for following page was skipped.

content/ja/guide/v10/forms.md

**Comment by 38elements** [comment]
Review for following pages were completed.

content/ja/guide/v10/hooks.md
content/ja/guide/v10/options.md

**Comment by 38elements** [comment]
Review for following pages were completed.

content/ja/guide/v10/progressive-web-apps.md
content/ja/guide/v10/refs.md
content/ja/guide/v10/server-side-rendering.md
content/ja/guide/v10/switching-to-preact.md
content/ja/guide/v10/tutorial.md

File: content/ja/index.md
Comment (38elements) [inline content/ja/index.md:13]: 敢えてこのままにしておく。

**Comment by eiiches** [comment]
I did an initial review with @38elements over Discord calls. We are planning to do a second review, going through all over again to make the translation more natural and easy to read. After that, I think this PR will be ready to be merged.

We skipped a review for content/ja/guide/v10/forms.md because the content is based on #591 which is not merged yet.

File: CONTRIBUTING.md
Comment (38elements) [inline CONTRIBUTING.md:1]: @developit @marvinhagemeister 
I added this document.

**Review by eiiches — APPROVED** [review]
We've completed a second review except for content/ja/guide/v10/forms.md. The overall quality of the translation is pretty good and I think it's ready to be published.

File: content/ja/index.md
Comment (38elements) [inline content/ja/index.md:49]: @developit @marvinhagemeister 
I decided Preact is プリアクト in Japanese.

**Comment by 38elements** [comment]
@eiiches 

Thank you for review.
The half of this contribution is yours.

File: CONTRIBUTING.md
Comment (developit) [reply CONTRIBUTING.md:1]: Excellent.

File: content/ja/index.md
Comment (developit) [reply content/ja/index.md:49]: I actually love this!! Google tells me the english pronunciation is approximately "Puriakuto", which is awesome. It also seems to nicely translate _back_ to english ([example](https://translate-prod.sandbox.google.com/?sl=ja&tl=en&text=%E3%83%97%E3%83%AA%E3%82%A2%E3%82%AF%E3%83%88&op=translate)).

**Comment by developit** [comment]
This is amazing work. Absolutely amazing.

@38elements I took your suggestion and added a [CODEOWNERs file](https://github.com/preactjs/preact-www/blob/master/.github/CODEOWNERS) that will wait for your approval on any `/ja/` translation updates.

Again, thank you both so so much.

**Comment by 38elements** [comment]
Thank you for CODEOWNERS.
I am very delighted at this.

**Pull request merged**
